[FTBFS] bolt

```
[42/50] Compiling C object 'test-common@exe/tests_test-common.c.o'.
FAILED: test-common@exe/tests_test-common.c.o 
cc -Itest-common@exe -I. -I.. -I../tests -I../common -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/gio-unix-2.0 -flto -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=gnu99 -fstack-protector-strong -Waggregate-return -Wunused -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wempty-body -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Winit-self -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-discarded-qualifiers -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wold-style-definition -Woverride-init -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wwrite-strings -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pthread  -MD -MQ 'test-common@exe/tests_test-common.c.o' -MF 'test-common@exe/tests_test-common.c.o.d' -o 'test-common@exe/tests_test-common.c.o' -c ../tests/test-common.c
In file included from /usr/lib/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from ./bolt-enum-types.h:6,
                 from ../common/bolt-enums.h:24,
                 from ../tests/test-common.c:23:
/usr/include/glib-2.0/glib/gmacros.h:480:43: error: redefinition of 'glib_autoptr_cleanup_GEnumClass'
 #define _GLIB_AUTOPTR_FUNC_NAME(TypeName) glib_autoptr_cleanup_##TypeName
                                           ^~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:500:36: note: in expansion of macro '_GLIB_AUTOPTR_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_FUNC_NAME(TypeName) (TypeName **_ptr) { if (*_ptr) (func) (*_ptr); }         \
                                    ^~~~~~~~~~~~~~~~~~~~~~~
../tests/test-common.c:45:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GEnumClass, g_type_class_unref);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:480:43: note: previous definition of 'glib_autoptr_cleanup_GEnumClass' was here
 #define _GLIB_AUTOPTR_FUNC_NAME(TypeName) glib_autoptr_cleanup_##TypeName
                                           ^~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:500:36: note: in expansion of macro '_GLIB_AUTOPTR_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_FUNC_NAME(TypeName) (TypeName **_ptr) { if (*_ptr) (func) (*_ptr); }         \
                                    ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gobject/gobject-autocleanups.h:25:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(GEnumClass, g_type_class_unref)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:482:48: error: redefinition of 'glib_listautoptr_cleanup_GEnumClass'
 #define _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) glib_listautoptr_cleanup_##TypeName
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:501:36: note: in expansion of macro '_GLIB_AUTOPTR_LIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) (GList **_l) { g_list_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../tests/test-common.c:45:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GEnumClass, g_type_class_unref);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:482:48: note: previous definition of 'glib_listautoptr_cleanup_GEnumClass' was here
 #define _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) glib_listautoptr_cleanup_##TypeName
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:501:36: note: in expansion of macro '_GLIB_AUTOPTR_LIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) (GList **_l) { g_list_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gobject/gobject-autocleanups.h:25:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(GEnumClass, g_type_class_unref)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:484:49: error: redefinition of 'glib_slistautoptr_cleanup_GEnumClass'
 #define _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) glib_slistautoptr_cleanup_##TypeName
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:502:36: note: in expansion of macro '_GLIB_AUTOPTR_SLIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) (GSList **_l) { g_slist_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../tests/test-common.c:45:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GEnumClass, g_type_class_unref);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:484:49: note: previous definition of 'glib_slistautoptr_cleanup_GEnumClass' was here
 #define _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) glib_slistautoptr_cleanup_##TypeName
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:502:36: note: in expansion of macro '_GLIB_AUTOPTR_SLIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) (GSList **_l) { g_slist_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gobject/gobject-autocleanups.h:25:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(GEnumClass, g_type_class_unref)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:480:43: error: redefinition of 'glib_autoptr_cleanup_GFlagsClass'
 #define _GLIB_AUTOPTR_FUNC_NAME(TypeName) glib_autoptr_cleanup_##TypeName
                                           ^~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:500:36: note: in expansion of macro '_GLIB_AUTOPTR_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_FUNC_NAME(TypeName) (TypeName **_ptr) { if (*_ptr) (func) (*_ptr); }         \
                                    ^~~~~~~~~~~~~~~~~~~~~~~
../tests/test-common.c:46:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GFlagsClass, g_type_class_unref);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:480:43: note: previous definition of 'glib_autoptr_cleanup_GFlagsClass' was here
 #define _GLIB_AUTOPTR_FUNC_NAME(TypeName) glib_autoptr_cleanup_##TypeName
                                           ^~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:500:36: note: in expansion of macro '_GLIB_AUTOPTR_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_FUNC_NAME(TypeName) (TypeName **_ptr) { if (*_ptr) (func) (*_ptr); }         \
                                    ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gobject/gobject-autocleanups.h:26:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(GFlagsClass, g_type_class_unref)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:482:48: error: redefinition of 'glib_listautoptr_cleanup_GFlagsClass'
 #define _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) glib_listautoptr_cleanup_##TypeName
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:501:36: note: in expansion of macro '_GLIB_AUTOPTR_LIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) (GList **_l) { g_list_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../tests/test-common.c:46:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GFlagsClass, g_type_class_unref);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:482:48: note: previous definition of 'glib_listautoptr_cleanup_GFlagsClass' was here
 #define _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) glib_listautoptr_cleanup_##TypeName
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:501:36: note: in expansion of macro '_GLIB_AUTOPTR_LIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) (GList **_l) { g_list_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gobject/gobject-autocleanups.h:26:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(GFlagsClass, g_type_class_unref)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:484:49: error: redefinition of 'glib_slistautoptr_cleanup_GFlagsClass'
 #define _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) glib_slistautoptr_cleanup_##TypeName
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:502:36: note: in expansion of macro '_GLIB_AUTOPTR_SLIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) (GSList **_l) { g_slist_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../tests/test-common.c:46:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GFlagsClass, g_type_class_unref);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:484:49: note: previous definition of 'glib_slistautoptr_cleanup_GFlagsClass' was here
 #define _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) glib_slistautoptr_cleanup_##TypeName
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:502:36: note: in expansion of macro '_GLIB_AUTOPTR_SLIST_FUNC_NAME'
   static G_GNUC_UNUSED inline void _GLIB_AUTOPTR_SLIST_FUNC_NAME(TypeName) (GSList **_l) { g_slist_free_full (*_l, (GDestroyNotify) (void(*)(void)) func); } \
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gobject/gobject-autocleanups.h:26:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(GFlagsClass, g_type_class_unref)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[43/50] Linking target test-exported.
ninja: build stopped: subcommand failed.
```

--- a/srcpkgs/bolt/patches/a9fda276415abb4c0ae7031271c2d5703c435086.patch
+++ b/srcpkgs/bolt/patches/a9fda276415abb4c0ae7031271c2d5703c435086.patch
@@ -1,0 +1,32 @@
+From a9fda276415abb4c0ae7031271c2d5703c435086 Mon Sep 17 00:00:00 2001
+From: Christian Kellner <christian@kellner.me>
+Date: Fri, 15 Jun 2018 15:43:54 +0200
+Subject: [PATCH] test: common: no enum auto cleanup for glib > 2.57
+
+Newer glib (starting with 2.57) have gained support for autoptr
+cleanup functions for GEnumClass and GFlagsClass, so we don't
+need to define our own ones anymore.
+This change was already done to bolt-enums.c, but also is needed
+for the test.
+---
+ tests/test-common.c | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/tests/test-common.c b/tests/test-common.c
+index c44d8ae..eb50a81 100644
+--- tests/test-common.c
++++ tests/test-common.c
+@@ -43,8 +43,10 @@
+ #include <sys/stat.h>
+ #include <unistd.h> /* unlinkat */
+ 
++#if !GLIB_CHECK_VERSION (2, 57, 0)
+ G_DEFINE_AUTOPTR_CLEANUP_FUNC (GEnumClass, g_type_class_unref);
+ G_DEFINE_AUTOPTR_CLEANUP_FUNC (GFlagsClass, g_type_class_unref);
++#endif
+ 
+ static void
+ cleanup_dir (DIR *d)
+-- 
+2.18.0
+

